walk/walk.go

```diff
@@ -356,3 +359,42 @@ func resolveFileInfo(wc *walkConfig, dir, rel string, ent fs.DirEntry) fs.DirEnt
 	}
 	return fs.FileInfoToDirEntry(fi)
 }
+
+func buildTrie(c *config.Config, isIgnored isIgnoredFunc) (*pathTrie, error) {
+	trie := &pathTrie{}
+
+	eg := errgroup.Group{}
+	eg.SetLimit(100)
+	eg.Go(func() error {
+		return walkDir(c.RepoRoot, "", &eg, isIgnored, trie)
+	})
+
+	return trie, eg.Wait()
+}
+
+// walkDir recursively descends path, calling walkDirFn.
+func walkDir(root, rel string, eg *errgroup.Group, isIgnored isIgnoredFunc, trie *pathTrie) error {
+	dirs, err := os.ReadDir(path.Join(root, rel))
+	if err != nil {
+		return err
+	}
+
+	for _, d1 := range dirs {
+		name1 := d1.Name()
+		path1 := path.Join(rel, name1)
+
+		// Ignore .git, empty names and ignored paths
```

Comment:
Should the .git check be only in the repo root? Do we actually want to support nested git repos?

Reply:
I have no idea, I'm just copying what was done before to a place that I think is more appropriate 🤷 

Functionally I think this is the same as before...